study1/week3/reduce-array-size-to-the-half.java

```diff
@@ -0,0 +1,30 @@
+// https://leetcode.com/problems/reduce-array-size-to-the-half/
+// Time Complexity: O(n), n: arr.length
+// Space Complexity: O(n), n: arr.length
+// In order that at least half of the integers of the array are removed, the most numerous integer has to be removed first (The more integer are removed, the faster can the array be smaller)
+class Solution {
+    public int minSetSize(int[] arr) {
+        int setSize = 0;
+        HashMap<Integer, Integer> integerMap = new HashMap<>();
+
+        // put value into map
+        for (int a : arr) {
+            integerMap.put(a, integerMap.getOrDefault(a, 0) + 1);
+        }
+
+        // in a descending order
+        List<Integer> keyList = new ArrayList<>(integerMap.keySet());
```

Comment:
So the information of the key itself loses meaning after counting the occurrences, because we only need how many count each key has, but the key doesn't really matter. So instead, you can try to do the following instead : 
```java
List<Integer> counts = new ArrayList<>(integerMap.values());
Collections.sort(counts);
```